src/main/java/uk/gov/hmcts/ccd/domain/service/common/AccessControlService.java

```diff
@@ -228,17 +240,30 @@ public CaseEventTrigger setReadOnlyOnCaseViewFieldsIfNoAccess(final CaseEventTri
                     CaseField field = caseFieldOpt.get();
                     if (!hasAccessControlList(userRoles, access, field.getAccessControlLists())) {
                         caseViewField.setDisplayContext(READONLY);
+                        setReadonlyOnWizardPageFieldDisplayContext(caseEventTrigger.getWizardPages(), caseViewField.getId());
                     }
                     if (field.isCompound()) {
                         setChildrenAsReadOnlyIfNoAccess(caseEventTrigger.getWizardPages(), field.getId(), field, access, userRoles, caseViewField);
                     }
                 } else {
                     caseViewField.setDisplayContext(READONLY);
+                    setReadonlyOnWizardPageFieldDisplayContext(caseEventTrigger.getWizardPages(), caseViewField.getId());
                 }
             });
         return caseEventTrigger;
     }
 
+    private void setReadonlyOnWizardPageFieldDisplayContext(List<WizardPage> wizardPages,
+                                                            String caseViewFieldId) {
+        wizardPages.forEach(wizardPage -> {
+            wizardPage.getWizardPageFields().stream()
+                .filter(wizardPageField -> wizardPageField.getCaseFieldId().equals(caseViewFieldId)).collect(toList())
```

Comment:
we could introduce a method on wizardPage called `getField` and move this code there